nixos/modules/services/system/earlyoom.nix

```
@@ -178,6 +178,11 @@ in
       # from /etc/default/earlyoom.
       serviceConfig.EnvironmentFile = "";
 
+      # Workaround: Upstream DynamicUser=true prevents notifications being sent to D-Bus.
+      # - earlyoom: https://github.com/NixOS/nixpkgs/issues/374959
+      # - systemd: https://github.com/systemd/systemd/issues/22737
+      serviceConfig.DynamicUser = mkDefault (!cfg.enableNotifications);
```

After reading https://github.com/systemd/systemd/issues/22737, I can confirm notifications work with broker enabled.

```suggestion
      serviceConfig.DynamicUser = mkDefault (
        !cfg.enableNotifications || config.services.dbus.implementation != "dbus"
      );
```

Whether you go with `config.services.dbus.implementation != "dbus"` or `config.services.dbus.implementation == "broker"` is whether you believe the bug is specific to dbus or a fix specific to broker. Currently there's only two possibilities for `config.services.dbus.implementation` so they're the same. I guess I would be optimistic and hope that future implementations won't have this problem. And you could add a test that it works with broker, at least. And maybe add a little note on this workaround in the description of `enableNotifications`?

---

Updated.